scripts/kms/jwt_issuer_trust.sh

```diff
@@ -26,139 +43,176 @@ decode_jwt() {
-    \"idtyp\": \"$(echo "$DECODED_JWT" | jq -r '.idtyp')\", \
-    \"oid\": \"$(echo "$DECODED_JWT" | jq -r '.oid')\""
+    source $REPO_ROOT/scripts/ccf/member/add.sh
 
-  set +e
-}
+    # Default to --demo behaviour
+    if [[ $# -eq 0 ]]; then
+        JWKS=`jwt-issuer-get-jwks-from-file $JWT_ISSUER_WORKSPACE/private.pem`
+        JWT_CLAIMS=`jwt-issuer-get-policy-from-token $(. $JWT_ISSUER_WORKSPACE/fetch.sh && jwt_issuer_fetch)`
+    fi
 
-jwt-issuer-trust() {
-  set -e
-
-  issuer="demo"
-
-  # Parse command-line arguments
-  while [[ $# -gt 0 ]]; do
-      case "$1" in
-          --demo)
-              issuer="demo"
-              shift 1
-              ;;
-          --aad)
-              issuer="aad"
-              shift 1
-              ;;
-          *)
-              echo "Unknown parameter: $1"
-              exit 1
-              ;;
-      esac
-  done
-
-  if [[ "$issuer" == "demo" ]]; then
-    use_demo_issuer
-    set_jwt_issuer
-  elif [[ "$issuer" == "aad" ]]; then
-    use_aad_issuer
+    # Parse command-line arguments
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            --private-key-path)
+                JWKS=`jwt-issuer-get-jwks-from-file "$2"`
+                shift 2
+                ;;
+            --private-key-json)
+                JWKS=`jwt-issuer-get-jwks-from-json $2`
+                shift 2
+                ;;
+            --token)
+                JWT_CLAIMS=`jwt-issuer-get-policy-from-token $2`
+                shift 2
+                ;;
+            --demo)
+                JWKS=`jwt-issuer-get-jwks-from-file $JWT_ISSUER_WORKSPACE/private.pem`
+                JWT_CLAIMS=`jwt-issuer-get-policy-from-token $(. $JWT_ISSUER_WORKSPACE/fetch.sh && jwt_issuer_fetch)`
+                shift 1
+                ;;
+            --managed-identity)
+                CA_CERT_BUNDLE_NAME="Microsoft_AAD"
+                CA_CERT_BUNDLE="$(awk '{printf "%s\\n", $0}' $REPO_ROOT/governance/jwt/aad_cert)"
+                CA_CERT_BUNDLE_NAME_FIELD="\"ca_cert_bundle_name\": \"$CA_CERT_BUNDLE_NAME\","
+                JWKS=`jwt-issuer-get-jwks-from-json \
+                    $(curl https://login.microsoftonline.com/$(az identity show --ids $2 | jq -r ".tenantId")/discovery/v2.0/keys)`
+                JWT_CLAIMS=`jwt-issuer-get-policy-from-mi $2`
+                shift 2
+                ;;
+            --managed-identity-v1)
+                CA_CERT_BUNDLE_NAME="Microsoft_AAD"
+                CA_CERT_BUNDLE="$(awk '{printf "%s\\n", $0}' $REPO_ROOT/governance/jwt/aad_cert)"
+                CA_CERT_BUNDLE_NAME_FIELD="\"ca_cert_bundle_name\": \"$CA_CERT_BUNDLE_NAME\","
+                JWKS=`jwt-issuer-get-jwks-from-json \
+                    $(curl https://login.microsoftonline.com/$(az identity show --ids $2 | jq -r ".tenantId")/discovery/v2.0/keys)`
+                JWT_CLAIMS=`jwt-issuer-get-policy-from-mi-v1 $2`
+                shift 2
+                ;;
+            --current-user|--aad)
```

Comment:
Is ` --current-az-user` might be clearer?

Reply:
I don't have a strong opinion so happy to change it
https://github.com/microsoft/azure-privacy-sandbox-kms/pull/439/commits/4e0b87c02eaf69688b73c8f6eb9e8c6b4e7f55fc

Reply:
Resolving as Ken approved the PR and it blocks other work